tests/test_cases/test_external/test_external.py

```diff
@@ -205,6 +205,30 @@ def test_time_in_external(dut):
         raise TestFailure("Time has elapsed over external call")
 
 
+@cocotb.function
+def yield_to_next_clock(dut):
+    yield RisingEdge(dut.clk)
+
+
+def test_ext_instantaneous_return(dut):
+    yield_to_next_clock(dut)
+    return get_sim_time('ps')
+
+
+@cocotb.test(expect_fail=False)
+def test_instantaneous_return(dut):
+    """Test that the simulation time does not advance between the final
+    function call in the external and the calling coroutine continuing"""
+    period = 100
+    clk_gen = cocotb.fork(Clock(dut.clk, period).start())
+    yield Timer(10, 'ns')
+    ext_sim_time = yield cocotb.external(test_ext_instantaneous_return)(dut)
```

Comment:
I think you maybe just want a `cocotb.external` decorator on `test_ext_instantaneous_return`.